src/gosling-track/gosling-track.ts

```diff
@@ -764,38 +760,37 @@ function GoslingTrack(HGC: import('@higlass/types').HGC, ...args: any[]): any {
             // Increase the size of tiles by length
             this.tileSize = (this.tilesetInfo?.tile_size ?? 1024) * tiles.length;
 
-            let newData: Datum[] = [];
+            let merged: Datum[] = [];
 
-            tiles.forEach((t: any, i: number) => {
-                // Combine data
-                newData = [...newData, ...t.tileData];
+            tiles.forEach((tile, i) => {
+                const tileInfo = this.processedTileInfo[tile.tileId];
+                if (tileInfo) {
+                    // Combine data
+                    merged = [...merged, ...tileInfo.tabularData];
 
-                // Flag to force using only one tile
-                t.mergedToAnotherTile = i !== 0;
+                    // Flag to force using only one tile
+                    tileInfo.mergedToAnotherTile = i !== 0;
+                }
             });
 
-            tiles[0].gos.raw = newData;
+            const firstTileInfo = this.processedTileInfo[tiles[0].tileId];
+            firstTileInfo.tabularData = merged;
```

Comment:
Yes, that is correct – When `displace` transform is used, we combine all tabular data into the first tile so that we apply the transformation at once. We iterate through tiles to render each tile, but since `mergedToAnotherTile` is `true` in the rest of the tiles, the rendering is skipped.